packages/core/src/Stores/client-store.js

```diff
@@ -1224,12 +1223,26 @@ export default class ClientStore extends BaseStore {
      */
     @action.bound
     async init(login_new_user) {
-        const search = window.location.search;
-        const search_params = new URLSearchParams(search);
+        this.initial_search_params = window.location.search;
+        const search_params = new URLSearchParams(this.initial_search_params);
         const redirect_url = search_params?.get('redirect_url');
 
         this.setIsLoggingIn(true);
         const authorize_response = await this.setUserLogin(login_new_user);
+        if (this.initial_search_params) {
+            document.addEventListener('DOMContentLoaded', () => {
+                setTimeout(() => {
+                    // timeout is needed for the app to get all necessary query params from the full URL
+                    // before we hide them from LiveChat getting the URL from Window, particularly when initialized via HTML script on mobile
+                    history.replaceState(null, null, `/${window.location.hash}`);
+                }, 0);
+            });
+            window.onload = () => {
+                // restoring all the necessary query params in the URL:
+                history.replaceState(null, null, this.replacement_url);
+            };
```

Comment:
why do we need to remove the query params and restore them again?
If the query parameters are sensitive, we should completely remove them after reading them.  e.g. tokens
If the query parameters are not sensitive, I think it's fine if those are sent to livechat e.g. lang

Reply:
good point! thank you 🙂  I gave it some thought and decided to remove the token only without restoring it in the url.